doc: add FAQ (#3)
snyk-labs/nopp: README.md

`TypeError: Cannot assign to read only property` exception and cause DoS
vulnerability. Please make sure you have `uncaughtException` handler
implemented.

## FAQ

### Should I prefer `nopp` instead of the `--frozen-intrinsics` Node.js flag?

[`--frozen-intrinsics`](https://nodejs.org/docs/latest-v17.x/api/cli.html#--frozen-intrinsics) added in Node.js v11.12.0 and currently has experimental stability level.

The main purpose of the flag is exactly the same as of this package – to protect runtime from unintended modifications of prototypes.

> We believe there is number of reasons why you may prefer using `nopp`:

Reviewer: ```suggestion
We believe there are numerous reasons why you may prefer using `nopp`:
```

1. You control when to import the package hence when to freeze prototypes. In many cases application actually modify prototypes a bit to add some tweaks or polyfills. In such cases usage of `--frozen-intrinsics` will be not possible without significant application code refactoring. Unlike `nopp` which should be imported after all other packages and in most of the cases cause no backward compatibility issues.

> 2. `nopp` is also applicable for client-side applications. You may prefer to use it for consistency between.

Reviewer: consistency between backend and frontend I assume you were aiming for?

```suggestion
2. `nopp` is also applicable for client-side applications. You may prefer to use it for consistency between backend Node.js code and client-side application code.
```

> 3. It gives you additional perks if you are Snyk customer. We are able to detect usage of `nopp` and ignore prototype pollution vulnerabilities in your application automatically.

Reviewer: ```suggestion
3. If you are a [Snyk user](https://snyk.io), we are able to detect the usage of the `nopp` package as part of your application and ignore prototype pollution vulnerabilities in your application code automatically, and help reduce the noise level of your overall security alerts.
```
